### PR TITLE
ci: add publish Docker workflow

### PR DESCRIPTION
**Pull request by lvarin**
- Add maven
- Add jdk8
- Add Dockerhub profile
- Changing Dockerhub repository
- Add Dockerfile option

### CODE REVIEW

File: .github/workflows/docker-build-publish.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        required: false
+        default: tesk-api
+
+
+jobs:
+  build-from-source:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout elixir-cloud-aai/tesk-api
+        uses: actions/checkout@v2
```
Comment (uniqueg) [inline .github/workflows/docker-build-publish.yml:21]: Please update to v3

File: .github/workflows/docker-build-publish.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+      
+      - name: Build with Maven
+        run: bash -c './mvnw clean verify'
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (uniqueg) [inline .github/workflows/docker-build-publish.yml:33]: Please update to v2

File: .github/workflows/docker-build-publish.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (uniqueg) [inline .github/workflows/docker-build-publish.yml:36]: Please update to v2

File: .github/workflows/docker-build-publish.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+      - name: Login to DockerHub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v1
```
Comment (uniqueg) [inline .github/workflows/docker-build-publish.yml:40]: Please update v2

File: .github/workflows/docker-build-publish.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v3
```
Comment (uniqueg) [inline .github/workflows/docker-build-publish.yml:47]: Please update to v4

File: .github/workflows/docker-build-publish.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        with:
+          images: elixircloud/tesk-api #${{ github.GITHUB_REF_TYPE }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{version}},value=v1.1.0
+      
+      - name: Build and push Docker images
+        uses: docker/build-push-action@v2
```
Comment (uniqueg) [inline .github/workflows/docker-build-publish.yml:55]: Please update to v3

**Review by uniqueg — CHANGES_REQUESTED** [review]
Thanks. Generally looking good, but CI generates warning due to security concerns with and depreciation of `set-output`. Updating all used actions should fix those issues.

image

File: .github/workflows/docker-build-publish.yml (hunk 6 above)
Comment (lvarin) [reply .github/workflows/docker-build-publish.yml]: done

File: .github/workflows/docker-build-publish.yml (hunk 5 above)
Comment (lvarin) [reply .github/workflows/docker-build-publish.yml]: done

File: .github/workflows/docker-build-publish.yml (hunk 4 above)
Comment (lvarin) [reply .github/workflows/docker-build-publish.yml:40]: done

File: .github/workflows/docker-build-publish.yml (hunk 3 above)
Comment (lvarin) [reply .github/workflows/docker-build-publish.yml]: done

File: .github/workflows/docker-build-publish.yml (hunk 2 above)
Comment (lvarin) [reply .github/workflows/docker-build-publish.yml]: done

File: .github/workflows/docker-build-publish.yml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+jobs:
+  build-from-source:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout elixir-cloud-aai/tesk-api
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
```
[the remark this answers is not in the document]
Comment (lvarin) [reply .github/workflows/docker-build-publish.yml]: done

File: .github/workflows/docker-build-publish.yml (hunk 1 above)
Comment (lvarin) [reply .github/workflows/docker-build-publish.yml]: done

**Comment by lvarin** [comment]
All warning seem to have disappeared.

**Review by uniqueg — APPROVED** [review]
Looks great :)

**Pull request merged**